airflow/decorators/task_group.py

```diff
@@ -61,7 +61,7 @@ def __attrs_post_init__(self):
     def _make_task_group(self, **kwargs) -> TaskGroup:
         return TaskGroup(**kwargs)
 
-    def __call__(self, *args, **kwargs) -> R:
+    def __call__(self, *args, **kwargs) -> Union[R, TaskGroup]:
```

Comment:
```
@task_group
def foo() -> BashOperator
    return BashOperator(task_id='a')

bash_op = foo()
```

Reply:
It maybe doesn't _need_ to be, but it feels like it should be.

The "Union" is because I couldn't find a way  of doing an overload on a class-level generic.

THe ideal is: If R is None return, then the return type of call is TaskGroup, else it's R